### PR TITLE
LC 2722 reporting export

### PR DESCRIPTION
**Pull request by jaylco**
- Add support for generating a data export request
- Refactor code and move some business logic to the backend
- Tweak some of the frontend presentation code

### CODE REVIEW

File: src/controllers/reportingController.ts [hunk 1]
```diff
@@ -47,7 +46,6 @@ export class ReportingController {
 					},
 					organisationListForTypeAhead: organisationChoices.typeaheadOrganisations,
 					showTypeaheadOption: userCanAccessMultipleOrganisations,
-					backButton: '/reporting'
```
Comment (neeraj-jain-cbo) [inline src/controllers/reportingController.ts:50]: If this line is deleted, the comma to the end of the previous line may need to be removed.

File: views/component/nav/template.njk [hunk 2]
```diff
@@ -12,7 +12,7 @@
                 <a class="nav-item govuk-link {% if originalUrl === '/content-management' %}active{% endif %}" href="/content-management">Admin</a>
             {% endif %}
             {% if identity.isReporter() %}
-                <a class="nav-item govuk-link {% if originalUrl === '/reporting' %}active{% endif %}" href="/reporting">Reporting</a>
+                <a class="nav-item govuk-link {% if originalUrl and originalUrl | startsWith('/reporting') %}active{% endif %}" href="/reporting">Reporting</a>
```
Comment (neeraj-jain-cbo) [inline views/component/nav/template.njk:15]: is it the correct syntax for the condition?

File: views/page/reporting/courseCompletions/choose-courses.html [hunk 3]
```diff
@@ -8,6 +8,8 @@
 
 {% extends "../../../component/Page.njk" %}
 
+{% set backButton = "/reporting/course-completions/choose-organisation" %}
```
Comment (neeraj-jain-cbo) [inline views/page/reporting/courseCompletions/choose-courses.html:11]: backButton value is set, but I can't see if it is used somewhere in this PR.

File: src/controllers/reportingController.ts (hunk 1 above)
Comment (jaylco) [reply src/controllers/reportingController.ts:50]: This is OK by Typescript standards (it's even required when using strict linting)

File: views/component/nav/template.njk (hunk 2 above)
Comment (jaylco) [reply views/component/nav/template.njk:15]: Yes, the startsWith function is a Nunjucks filter and is used in templates with the pipe syntax

File: views/page/reporting/courseCompletions/choose-courses.html (hunk 3 above)
Comment (jaylco) [reply views/page/reporting/courseCompletions/choose-courses.html:11]: backButton is a nunjucks block and can be found in the imported `page.njk` template (https://github.com/Civil-Service-Human-Resources/lpg-management/blob/ecf261eb74dcc8e731bb54020a508837a2f698b5/views/component/Page.njk#L78)

**Review by neeraj-jain-cbo — APPROVED** [review]

**Pull request merged**